File: SeongYeon/privacyPolicy.js
function solution(today, terms, privacies) {
  let answer = [];
  const termsObj = {};
  const todayDate = new Date(today);

  terms.forEach((term) => {
    const tuple = term.split(' ');
    termsObj[tuple[0]] = Number(tuple[1]);
  });

  privacies.forEach((privacy, idx) => {
    const tuple = privacy.split(' ');
    const expDate = new Date(tuple[0]);
    const term = termsObj[tuple[1]];
    expDate.setMonth(expDate.getMonth() + term);

    if (todayDate >= expDate) answer.push(idx + 1);
  });

  return answer;
}

File: jiwon/privacy_policy.py
def get_days(y, m, d, m2='0'):
    return int(y[2:]) * 28 * 12 + (int(m) + int(m2)) * 28 + int(d)


def solution(today, terms, privacies):
    answer = []
    today_days = get_days(*today.split('.'))
    terms_dic = dict(term.split(" ") for term in terms)

    for i, p in enumerate(privacies, start=1):
        date, term = p.split(' ')
        if today_days > get_days(*date.split('.'), terms_dic[term]) - 1:
            answer.append(i)
    return answer
